### PR TITLE
[ryu] Fix VS2022 build

### PR DESCRIPTION
**Pull request by sbrajchuk**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by jimwang118** [comment]
Pinging @sbrajchuk for response. Is work still being done for this PR?

**Comment by sbrajchuk** [comment]
Yes it is.

File: ports/ryu/portfile.cmake [hunk 1]
```diff
@@ -14,18 +14,31 @@ set(ENV{BAZEL_BIN_PATH} "${BAZEL}")
...
+    if (VCPKG_CRT_LINKAGE STREQUAL "dynamic")
+        set(USE_RUNTIME_LIBRARY "/MD")
+    elseif (VCPKG_CRT_LINKAGE STREQUAL "static")
+        set(USE_RUNTIME_LIBRARY "/MT")
+    endif ()
 endif ()
 
 vcpkg_execute_build_process(
-            COMMAND ${BAZEL} build --verbose_failures --strategy=CppCompile=standalone //ryu //ryu:ryu_printf
+            COMMAND ${BAZEL} build --cxxopt=${USE_RUNTIME_LIBRARY} --verbose_failures --strategy=CppCompile=standalone //ryu //ryu:ryu_printf
```
[the remark this answers is not in the document]
Comment (sbrajchuk) [reply ports/ryu/portfile.cmake:25]: thanks

**Comment by jimwang118** [comment]
If I install the port, how can I call the port header file and function in my test project?

**Comment by sbrajchuk** [comment]
```c++
#include <iostream>
#include <ryu/ryu.h>

int main() {
    char buf[25];
    double n = 3.9923000000000002;
    double n1 = 1.0000000000000001e-05;
    d2s_buffered(n, buf);
    std::cout << buf << std::endl;
    d2s_buffered(n1, buf);
    std::cout << buf << std::endl;
    return 0;
}
```

**Comment by jimwang118** [comment]
Ok, I suggest that you add a usage file in the port directory to provide an example of how to use it, so that other users can use this port.
Add the following to the usage file:
```
find_package(ryu CONFIG REQUIRED)
target_link_libraries(main PRIVATE RYU::ryu)
```

**Comment by jimwang118** [comment]
The usage has been tested successfully locally. 
```
find_package(ryu CONFIG REQUIRED)
target_link_libraries(main PRIVATE RYU::ryu)
```

**Comment by sbrajchuk** [comment]
Please do not merge this pull request at the moment. I have discovered some issues and need to recheck it.

**Comment by sbrajchuk** [comment]
Issues have been fixed.

File: ports/ryu/portfile.cmake [hunk 2]
```diff
@@ -6,35 +6,76 @@ vcpkg_from_github(
...
+    file(INSTALL ${SOURCE_PATH}/bazel-bin/ryu/libryu.a DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
     file(INSTALL ${SOURCE_PATH}/bazel-bin/ryu/libryu_printf.a DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
-endif()
+endif ()
 
 file(INSTALL ${SOURCE_PATH}/LICENSE-Boost DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 file(INSTALL ${SOURCE_PATH}/ryu/ryu.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/ryu/)
 file(INSTALL ${SOURCE_PATH}/ryu/ryu2.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/ryu/)
 file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/ryuConfig.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```
Comment (jimwang118) [inline ports/ryu/portfile.cmake:81]:
```suggestion
file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})

```
You need to add a blank line at the end of the file.

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/ryu/portfile.cmake (hunk 2 above)
Comment (sbrajchuk) [reply ports/ryu/portfile.cmake]: The line has been added.

**Review by jimwang118 — DISMISSED** [review]

**Pull request merged**